Description: Skip tests that use python-googleapi as it is not
             in Ubuntu main.
Author: Corey Bryant <corey.bryant@canonical.com>
Forwarded: not-needed

--- a/cinder/tests/unit/test_backup_google.py
+++ b/cinder/tests/unit/test_backup_google.py
@@ -25,20 +25,48 @@
 import os
 import shutil
 import tempfile
+import testtools
 import zlib
 
 import mock
 from oslo_utils import units
 
-from cinder.backup.drivers import google as google_dr
+try:
+    from cinder.backup.drivers import google as google_dr
+    google_dr_backup_gcs_bucket = google_dr.CONF.backup_gcs_bucket
+    google_dr_client = google_dr.client
+    google_dr_discovery = google_dr.discovery
+    google_dr_backup_driver = google_dr.GoogleBackupDriver
+except ImportError:
+    google_dr = None
+    google_dr_backup_gcs_bucket = None #'gcscinderbucket'
+    google_dr_client = None
+    google_dr_discovery = None
+    google_dr_backup_driver = None
 from cinder import context
 from cinder import db
 from cinder import exception
 from cinder.i18n import _
 from cinder import objects
 from cinder import test
-from cinder.tests.unit.backup import fake_google_client
-from cinder.tests.unit.backup import fake_google_client2
+try:
+    from cinder.tests.unit.backup import fake_google_client
+    from cinder.tests.unit.backup import fake_google_client2
+    fake_google_client_fakecreds = fake_google_client.FakeGoogleCredentials
+    fake_google_client_fakediscbuild = fake_google_client.FakeGoogleDiscovery.Build
+    fake_google_client_fakemediadown = fake_google_client.FakeGoogleMediaIoBaseDownload
+    fake_google_client2_fakecreds = fake_google_client2.FakeGoogleCredentials
+    fake_google_client2_fakediscbuild = fake_google_client2.FakeGoogleDiscovery.Build
+    fake_google_client2_fakemediadown = fake_google_client2.FakeGoogleMediaIoBaseDownload
+except ImportError:
+    fake_google_client = None
+    fake_google_client2 = None
+    fake_google_client_fakecreds = None
+    fake_google_client_fakediscbuild = None
+    fake_google_client_fakemediadown = None
+    fake_google_client2_fakecreds = None
+    fake_google_client2_fakediscbuild = None
+    fake_google_client2_fakemediadown = None
 
 
 class FakeMD5(object):
@@ -65,12 +93,12 @@
 
 
 def gcs_client(func):
-    @mock.patch.object(google_dr.client, 'GoogleCredentials',
-                       fake_google_client.FakeGoogleCredentials)
-    @mock.patch.object(google_dr.discovery, 'build',
-                       fake_google_client.FakeGoogleDiscovery.Build)
+    @mock.patch.object(google_dr_client, 'GoogleCredentials',
+                       fake_google_client_fakecreds)
+    @mock.patch.object(google_dr_discovery, 'build',
+                       fake_google_client_fakediscbuild)
     @mock.patch.object(google_dr, 'GoogleMediaIoBaseDownload',
-                       fake_google_client.FakeGoogleMediaIoBaseDownload)
+                       fake_google_client_fakemediadown)
     @mock.patch.object(hashlib, 'md5', FakeMD5)
     def func_wrapper(self, *args, **kwargs):
         return func(self, *args, **kwargs)
@@ -79,13 +107,13 @@
 
 
 def gcs_client2(func):
-    @mock.patch.object(google_dr.client, 'GoogleCredentials',
-                       fake_google_client2.FakeGoogleCredentials)
-    @mock.patch.object(google_dr.discovery, 'build',
-                       fake_google_client2.FakeGoogleDiscovery.Build)
+    @mock.patch.object(google_dr_client, 'GoogleCredentials',
+                       fake_google_client2_fakecreds)
+    @mock.patch.object(google_dr_discovery, 'build',
+                       fake_google_client2_fakediscbuild)
     @mock.patch.object(google_dr, 'GoogleMediaIoBaseDownload',
-                       fake_google_client2.FakeGoogleMediaIoBaseDownload)
-    @mock.patch.object(google_dr.GoogleBackupDriver,
+                       fake_google_client2_fakemediadown)
+    @mock.patch.object(google_dr_backup_driver,
                        '_generate_object_name_prefix',
                        FakeObjectName._fake_generate_object_name_prefix)
     @mock.patch.object(hashlib, 'md5', FakeMD5)
@@ -120,7 +148,7 @@
 
     def _create_backup_db_entry(self,
                                 volume_id=_DEFAULT_VOLUME_ID,
-                                container=google_dr.CONF.backup_gcs_bucket,
+                                container=google_dr_backup_gcs_bucket,
                                 parent_id=None,
                                 service_metadata=None):
 
@@ -141,6 +169,7 @@
         backup.create()
         return backup
 
+    @testtools.skipIf(not google_dr, "Skipped by Ubuntu")
     def setUp(self):
         super(GoogleBackupDriverTestCase, self).setUp()
         self.flags(backup_gcs_bucket='gcscinderbucket')
@@ -405,7 +434,7 @@
                           backup, self.volume_file)
 
     @gcs_client
-    @mock.patch.object(google_dr.GoogleBackupDriver, '_backup_metadata',
+    @mock.patch.object(google_dr_backup_driver, '_backup_metadata',
                        fake_backup_metadata)
     def test_backup_backup_metadata_fail(self):
         """Test of when an exception occurs in backup().
@@ -426,9 +455,9 @@
                           backup, self.volume_file)
 
     @gcs_client
-    @mock.patch.object(google_dr.GoogleBackupDriver, '_backup_metadata',
+    @mock.patch.object(google_dr_backup_driver, '_backup_metadata',
                        fake_backup_metadata)
-    @mock.patch.object(google_dr.GoogleBackupDriver, 'delete', fake_delete)
+    @mock.patch.object(google_dr_backup_driver, 'delete', fake_delete)
     def test_backup_backup_metadata_fail2(self):
         """Test of when an exception occurs in an exception handler.
 
@@ -511,7 +540,7 @@
         service.delete(backup)
 
     @gcs_client
-    @mock.patch.object(google_dr.GoogleBackupDriver, 'delete_object',
+    @mock.patch.object(google_dr_backup_driver, 'delete_object',
                        _fake_delete_object)
     def test_delete_without_object_prefix(self):
         volume_id = 'ee30d649-72a6-49a5-b78d-000000edb6b1'
